src/services/room-service.spec.js

```diff
@@ -331,4 +337,76 @@ describe('room-service', () => {
       expect(result).toBe(false);
     });
   });
+
+  describe('deleteRoom', () => {
+    const roomId = uniqueId.create();
+    beforeEach(async () => {
+      await roomStore.save({
+        _id: roomId,
+        name: 'my room',
+        access: ROOM_ACCESS_LEVEL.private,
+        owner: myUser._id,
+        members: [
+          {
+            userId: otherUser._id,
+            joinedOn: new Date()
+          }
+        ]
+      });
+    });
+
+    describe('when the room does not exist', () => {
+      it('should throw a not found exception', () => {
+        expect(() => sut.deleteRoom('badIdBadIdITellYou', myUser)).rejects.toThrow(NotFound);
+      });
+    });
+
+    describe('when another another user except the owner tries to delete', () => {
+      it('should throw a forbidden exception', () => {
+        expect(() => sut.deleteRoom(roomId, otherUser)).rejects.toThrow(Forbidden);
+      });
+    });
+
+    describe('when all goes well', () => {
+      let invitationDetails;
+      const cdnNameObject1 = `/rooms/${roomId}/object1`;
+      const cdnNameObject2 = `/rooms/${roomId}/object2`;
+
+      beforeEach(async () => {
+        sandbox.stub(cdn, 'listObjects').resolves([
+          { name: cdnNameObject1 },
+          { name: cdnNameObject2 }
+        ]);
+
+        sandbox.stub(cdn, 'deleteObjects');
+        invitationDetails = await sut.createOrUpdateInvitation({ roomId, email: otherUser.email, user: myUser });
+
+        await sut.deleteRoom(roomId, myUser);
+      });
+
+      afterEach(() => {
```

Comment:
The sandbox is global, so this `afterEach` should better be global, otherwise the sandbox works differently in this `describe` block then elsewhere.

Reply:
Fixed